portal/src/graphql/portal/PasswordlessAuthenticatorScreen.tsx

```diff
@@ -0,0 +1,213 @@
+import React, { useCallback, useMemo, useState } from "react";
+import { FormattedMessage } from "@oursky/react-messageformat";
+import { Label, Text } from "@fluentui/react";
+import { useParams } from "react-router-dom";
+import cn from "classnames";
+import produce from "immer";
+import deepEqual from "deep-equal";
+
+import { useUpdateAppConfigMutation } from "./mutations/updateAppConfigMutation";
+import { useAppConfigQuery } from "./query/appConfigQuery";
+import ShowError from "../../ShowError";
+import ShowLoading from "../../ShowLoading";
+import CodeEditor from "../../CodeEditor";
+import { PortalAPIAppConfig } from "../../types";
+import { clearEmptyObject } from "../../util/misc";
+import ButtonWithLoading from "../../ButtonWithLoading";
+import NavigationBlockerDialog from "../../NavigationBlockerDialog";
+
+import styles from "./PasswordlessAuthenticatorScreen.module.scss";
+
+interface PasswordlessAuthenticatorScreenState {
+  emailHtmlTemplate: string;
+  emailPlainTextTemplate: string;
+  smsTemplate: string;
+}
+
+function constructStateFromAppConfig(
+  /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
+  _appConfig: PortalAPIAppConfig | null
+): PasswordlessAuthenticatorScreenState {
+  return {
+    emailHtmlTemplate: "", // TODO: handle email template
+    emailPlainTextTemplate: "", // TODO: handle email template
+    smsTemplate: "", // TODO: handle sms template
+  };
+}
+
+function constructAppConfigFromState(
+  rawAppConfig: PortalAPIAppConfig,
+  /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
+  _initialScreenState: PasswordlessAuthenticatorScreenState,
+  /* eslint-disable-next-line @typescript-eslint/no-unused-vars */
+  _screenState: PasswordlessAuthenticatorScreenState
+): PortalAPIAppConfig {
+  const newAppConfig = produce(rawAppConfig, (draftConfig) => {
+    // TODO: update email template
+    // TODO: update sms template
+
+    clearEmptyObject(draftConfig);
+  });
+
+  return newAppConfig;
+}
+
+const PasswordlessAuthenticatorScreen: React.FC = function PasswordlessAuthenticatorScreen() {
+  const { appID } = useParams();
+
+  const {
+    updateAppConfig,
+    loading: updatingAppConfig,
+    error: updateAppConfigError,
+  } = useUpdateAppConfigMutation(appID);
+  const { loading, error, data, refetch } = useAppConfigQuery(appID);
+  const { effectiveAppConfig, rawAppConfig } = useMemo(() => {
+    const node = data?.node;
+    return node?.__typename === "App"
```

Comment:
Just a reminder on #398, let's fix it ASAP.

Reply:
Fixed